lexer/impl/lexer.cpp

```diff
@@ -9,315 +11,163 @@
-                        while (index < word.length() && (std::isalpha(word[index]) || std::isdigit(word[index]) || word[index] == '_'))
-                        {
-                            name.push_back(word[index++]);
-                        }
-                        val = trie.Find(name).value_or(TokenType::InvalidToken);
-                        if (val == TokenType::InvalidToken)
-                        {
-                            result.push_back({TokenType::Identifier, lineNum, name});
-                        }
-                        else
-                        {
-                            result.push_back({val, lineNum, sign});
-                        }
-                    }
-                    else
-                    { // non-valid construction or space
-                        // std::cout << "INVALID CHAR" << c << "\n";
-                        if (!std::isspace(c))
-                        {
-                            result.push_back({TokenType::InvalidToken, lineNum, toStr(c)});
-                        }
-                    }
-
-                    break;
-                }
-            }
+    if (std::isdigit(m_buf[m_pos])) {
+
+        auto truncLen = skipWhile(m_pos, Lexer::isdigit, 1);
+
+        size_t fracLen = 0;
+        if (m_pos + truncLen < m_buf.size() && m_buf[m_pos + truncLen] == '.') {
+            fracLen = skipWhile(m_pos + truncLen + 1, Lexer::isdigit);
         }
-        return 0;
-    }
 
-    /**
-     * Processing a line of riddle code. It updates the 'result' vector of tokens
-     * 
-     * @param line, string with line
-    */
-    void processLine(std::string line)
-    {
-        lineNum++;
-        std::istringstream iss(line);
-        std::string word;
-        while (std::getline(iss, word, ' '))
-        {
-            if (processWord(word) == -1)
-            { //if get a single-line comment
-                return;
-            }
+        if (fracLen != 0) {
+            truncLen += 1 + fracLen;
+            tok.type = TokenType::RealLiteral;
+        } else {
+            tok.type = TokenType::IntegerLiteral;
         }
-        result.push_back({TokenType::NewLine, lineNum, "\n"});
-    }
-    // ==================================PUBLIC================================
-    /**
- * Gets the trie from the lexer's namespace. 
- * Method should be used for testing and debugging only
- * 
- * @return trie, used during parsing (debugging purposes)
-*/
-    common::Trie<TokenType> getTrie()
-    {
-        initTrie();
-        return trie;
-    }
 
-    std::vector<Token> splitLine(std::string line)
-    {
-        initTrie();
-        processLine(line);
-        return result;
+        tok.image = m_buf.substr(m_pos, truncLen);
+
+        m_pos += truncLen;
+
+        return tok;
     }
 
-    void freeResult()
-    {
-        result = {};
+    if (m_buf[m_pos] == '\n') {
+        tok.type = TokenType::NewLine;
+        tok.image = "\n";
+
+        m_pos++;
```

Comment:
How about something that skips all whitespace after it (including other new lines)?

Reply:
Might make the parser's job a little easier, but whatever